library/std/canon.qs

```diff
@@ -476,4 +476,43 @@ namespace Microsoft.Quantum.Canon {
         }
     }
 
+    /// # Summary
+    /// Applies Quantum Fourier Transform (QFT) to a little-endian quantum register.
+    ///
+    /// # Description
+    /// Applies QFT to a little-endian register `qs` of length n
+    /// containing |x₁⟩⊗|x₂⟩⊗…⊗|xₙ⟩. The qs[0] contains the
+    /// least significant bit xₙ. The state of qs[0] becomes
+    /// (|0⟩+𝑒^(2π𝑖[0.xₙ])|1⟩)/sqrt(2) after the operation.
+    ///
+    /// # Input
+    /// ## qs
+    /// Quantum register in a little-endian format to which the QFT is applied.
+    ///
+    /// # Reference
+    ///  - [Quantum Fourier transform](https://en.wikipedia.org/wiki/Quantum_Fourier_transform)
+    operation ApplyQFT (qs : Qubit[]) : Unit is Adj + Ctl {
+        let length = Length(qs);
+        Fact(length >= 1, "ApplyQFT: Length(qs) must be at least 1.");
+        for i in length-1..-1..0 {
+            H(qs[i]);
+            for j in 0..i-1 {
+                Controlled R1Frac([qs[i]], (1, j+1, qs[i-j-1]));
+            }
+        }
+    }
+
+    /// # Summary
+    /// Uses SWAP gates to reverse the order of the qubits in a register.
+    ///
+    /// # Input
+    /// ## register
+    /// The qubits order of which should be reversed using SWAP gates
+    operation SwapReverseRegister (register : Qubit[]) : Unit is Adj + Ctl {
+        let length = Length(register);
+        for i in 0 .. length/2 - 1 {
+            SWAP(register[i], register[(length - i) - 1]);
+        }
+    }
```

Comment:
One more question, out of curiosity: why would someone use this to reverse the order of a qubit array, costing actual `SWAP` invocations, when they could just reverse the order via slicing with `register[...-1...]`?

Reply:
Slicing won't change the state of qubits, only reindex them. So if you want to actually set the state of specific qubit, you'll have to use SwapReverseRegister.
An example with two variables is probably easier to understand. If you have qubits named a and b, and after transformation X you got the state you wanted for a in b and vice versa, one option you have is to use the name 'a' in your program where you wanted the state of 'b'. But such code re-write isn't always possible. For example, if the code that runs before X and after X is the same.
This could be optimized on a different level. I expect actual quantum computer or simulator to have a mapping between qubits in the program to underlying qubits on the device. SWAP would just update that mapping performing no actual quantum operations.